synapse/storage/databases/state/bg_updates.py

```diff
@@ -93,158 +93,104 @@ def _get_state_groups_from_groups_txn(
 
         results: Dict[int, MutableStateMap[str]] = {group: {} for group in groups}
 
-        if isinstance(self.database_engine, PostgresEngine):
-            # Temporarily disable sequential scans in this transaction. This is
-            # a temporary hack until we can add the right indices in
-            txn.execute("SET LOCAL enable_seqscan=off")
-
-            # The below query walks the state_group tree so that the "state"
-            # table includes all state_groups in the tree. It then joins
-            # against `state_groups_state` to fetch the latest state.
-            # It assumes that previous state groups are always numerically
-            # lesser.
-            # This may return multiple rows per (type, state_key), but last_value
-            # should be the same.
-            sql = """
-                WITH RECURSIVE sgs(state_group) AS (
-                    VALUES(?::bigint)
-                    UNION ALL
-                    SELECT prev_state_group FROM state_group_edges e, sgs s
-                    WHERE s.state_group = e.state_group
-                )
-                %s
-            """
+        # Temporarily disable sequential scans in this transaction. This is
+        # a temporary hack until we can add the right indices in
+        txn.execute("SET LOCAL enable_seqscan=off")
+
+        # The below query walks the state_group tree so that the "state"
+        # table includes all state_groups in the tree. It then joins
+        # against `state_groups_state` to fetch the latest state.
+        # It assumes that previous state groups are always numerically
+        # lesser.
+        # This may return multiple rows per (type, state_key), but last_value
+        # should be the same.
+        sql = """
+            WITH RECURSIVE sgs(state_group) AS (
+                VALUES(CAST(? AS bigint))
```

Comment:
The only change to this code here is changing the cast to be compatible between SQLite and Postgres.

SQLite chokes on the previous `?::bigint` syntax